src/dbg/exprfunc.cpp

```diff
@@ -411,4 +412,59 @@ namespace Exprfunc
         SetContextDataEx(hActiveThread, UE_CIP, cip);
         return cip;
     }
+
+    duint MatchStrA(duint src, duint patstr)
+    {
+        const int MAX_SIZE = 300;
+        char dbgMemstr[MAX_SIZE] = {0};
+        duint step = sizeof(char);
+        for(int i = 0; i < MAX_SIZE; i++)
+        {
+            if(!DbgMemIsValidReadPtr(src + (i * step)))
+            {
+                dprintf("matchstrA Error reading [0x%x+0x%.2x*1]\n", src, i);
+                return 0;
+            }
+
+            DbgMemRead(src + (i * step), dbgMemstr + i, step);
+            if(*(dbgMemstr + i) == '\x00')
+                break;
+        }
+
+        std::regex reg1((char*)patstr, std::regex::icase);

```

Comment:
I was talking about this cast... If you use `matchstrA(rsp, 123)` it will be `(char*)123` or did I miss something?

Reply:
matchstrA(rsp, 123) is the wrong way to use, if you mean to use the string "123", it should be like this
matchstrA(rsp, #:0"123")

Reply:
I know it’s the wrong way to use it, but if you use it wrong anyway because of a mistake, will it crash?

Reply:
Oh, I will add a function to avoid this error.

Reply:
What about using C++ style cast?

```cpp
const std::regex reg1(reinterpret_cast<char*>(patstr), std::regex::icase);
```

Reply:
The cast probably won’t compile and if it does will still crash at runtime :)

The only solution here is to create a global string pool and let the patstr be the index in this pool...